packages/promotion/src/models/promotion-rule.ts

```diff
@@ -0,0 +1,83 @@
+import { PromotionRuleOperator } from "@medusajs/types"
+import { PromotionUtils, generateEntityId } from "@medusajs/utils"
+import {
+  BeforeCreate,
+  Cascade,
+  Collection,
+  Entity,
+  Enum,
+  Index,
+  ManyToMany,
+  OnInit,
+  OneToMany,
+  OptionalProps,
+  PrimaryKey,
+  Property,
+} from "@mikro-orm/core"
+import ApplicationMethod from "./application-method"
+import Promotion from "./promotion"
+import PromotionRuleValue from "./promotion-rule-value"
+
+type OptionalFields = "description"
+type OptionalRelations = "values" | "promotions"
+
+@Entity()
+export default class PromotionRule {
```

Comment:
todo: To be consistent with other module models, we should use type `| null` and `?` for nullable properties.